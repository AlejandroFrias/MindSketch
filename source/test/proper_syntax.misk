

# Comments for the Translator Object

#    Should handle multiple lines and preserve spacing
# An be parsable even with lines seperating them

Title should handle spaces and Numbers23:


# Parser Objects should get comments
PARSER START
words to parse y and $START and $END
PARSER END

# As well as Code Snippets

#    And the multi-line thing

CODE START: java
for(int i = ${1:$START}; i < ${2:$END}; i++) {
    $0
}
CODE END

# Multiple Translator Objects should be able to be parsed
# And they can have the same name without overwriting parsers
# But it could overwrite code snippets or add to them depending on 
# the language of the snippet
Title should handle spaces and Numbers23:

PARSER START 
whatever (i|do)? $START and $END
PARSER END

CODE START: python
for x in xrange(${1:$START}, ${2:$END}):
    ${0:pass}
CODE END
