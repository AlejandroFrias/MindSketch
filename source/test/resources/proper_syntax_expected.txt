# Translator Object: Title should handle spaces and Numbers23
# Comments for the Translator Object
#    Should handle multiple lines and preserve spacing
# An be parsable even with lines seperating them
# Multiple Translator Objects should be able to be parsed
# And they can have the same name without overwriting parsers
# But it could overwrite code snippets or add to them depending on 
# the language of the snippet

# Parser Objects should get comments
parsers.add('Title should handle spaces and Numbers23', (u'words', u'(and)', u'(some groups|groups of a kind|groups)', u'to', u'be', u'parsed', u'with', u'(?P<START>.*?)', u'and', u'(?P<END>.*?)', u'variables',))
parsers.add('Title should handle spaces and Numbers23', (u'whatever', u'(i|do)', u'(?P<START>.*?)', u'and', u'(?P<END>.*?)',))
# As well as Code Snippets
#    And the multi-line thing
code_snippets.add('Title should handle spaces and Numbers23', 'java', """for(int i = ${{1:{0[START]}}}; i < ${{2:{0[END]}}}; i++) {{
    $0
}}""")
code_snippets.add('Title should handle spaces and Numbers23', 'python', """for x in xrange(${{1:{0[START]}}}, ${{2:{0[END]}}}):
    ${{0:pass}}""")

